src/qt/walletmodel.cpp

```diff
@@ -594,7 +594,7 @@ WalletModel::UnlockContext WalletModel::requestUnlock(bool fForMixingOnly)
     EncryptionStatus encStatusNew = getEncryptionStatus();
 
     // Wallet was locked, user requested to unlock it for mixing and failed to do so
-    bool fMixingUnlockFailed = fForMixingOnly && !(encStatusNew == UnlockedForMixingOnly);
+    bool fMixingUnlockFailed = fForMixingOnly && !(encStatusNew == UnlockedForMixingOnly || encStatusNew == Unlocked);
```

Comment:
Why not `fForMixingOnly && encStatusNew != UnlockedForMixingOnly && encStatusNew != Unlocked`? Much easier to read IMHO

Reply:
Hmm... On another thought that's actually equal to `fForMixingOnly && encStatusNew == Locked`.. but there is no reason to check this because `fInvalid` already has `encStatusNew == Locked` as a part of the condition. The whole situation seems a bit illogical now that I think more of it tbh. Should we even allow user to unlock wallet _not_ for mixing this way? Probably not...